tests/pytests/test_triplet.py

```diff
@@ -0,0 +1,44 @@
+import numpy as np
+import pytest
+import time
+
+import psi4
+
+@pytest.mark.quick
+def test_triplet_speedup():
+    a = np.random.rand(1000, 100);
+    b = np.random.rand(100, 1000);
+    c = np.random.rand(1000, 100);
+
+    A = psi4.core.Matrix.from_array(a)
+    B = psi4.core.Matrix.from_array(b)
+    C = psi4.core.Matrix.from_array(c)
+
+    start = time.time()
+    R = psi4.core.triplet(A, B, C, False, False, False)
+    time1 = time.time() - start
+
+    start = time.time()
+    S = psi4.core.doublet(A, B, False, False)
+    S = psi4.core.doublet(S, C, False, False)
+    time2 = time.time() - start
+
+    assert (time1 < time2)
+    assert psi4.compare_matrices(R, S, 8, "linalg::triplet speedup test")
+
+@pytest.mark.quick
```

Comment:
Make sure to add test case with irreps. Especially cases where irrep sizes are zero.

Reply:
I noticed opt_einsum is among the deps of the open-source formulation of adcc (https://anaconda.org/adcc/adcc/files), so now it's in the psi4 dependency stream.

Reply:
good to add an irreped case here, since that figures in to your (AB)C vs A(BC) switch. an example of making such a `Matrix` is https://github.com/psi4/psi4/blob/master/tests/numpy-array-interface/input.dat#L143 .

Also, add the name of this test file https://github.com/psi4/psi4/blob/master/psi4/CMakeLists.txt#L308 so it installs.

Reply:
Done! :)